api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java

```diff
@@ -593,6 +588,23 @@ public void mergePatients(Patient preferred, Patient notPreferred) throws APIExc
 		Context.getPersonService().savePersonMergeLog(personMergeLog);
 	}
 	
+	private void requireNoActiveOrderOfSameType(Patient patient1, Patient patient2) {
+		String messageKey = "Patient.merge.cannotHaveSameTypeActiveOrders";
+		List<Order> ordersByPatient1 = Context.getOrderService().getAllOrdersByPatient(patient1);
+		List<Order> ordersByPatient2 = Context.getOrderService().getAllOrdersByPatient(patient2);
+		ordersByPatient1.forEach((Order order1) -> {
+			ordersByPatient2.forEach((Order order2) -> {
+				if (order1.isActive() && order2.isActive() && order1.getOrderType().equals(order2.getOrderType())) {
```

Comment:
We have discussed some of the scenarios of patient merge in this [talk](https://talk.openmrs.org/t/patient-merge-functionality/9310).
I want to highlight this [part](https://talk.openmrs.org/t/patient-merge-functionality/9310/16).
We have decided the following:
 We will check only whether the patients have active orders of SAME ORDER TYPE or not.
 We will not check whether the patients have active orders of SAME CONCEPT or not, as of now.

\*I hope this is clear and forgive me for the previous incomprehensible comment.